include/rapidjson/document.h

```diff
@@ -1235,8 +1235,17 @@ class GenericValue {
             // return NullValue;
 
             // Use static buffer and placement-new to prevent destruction
-            static GenericValue buffer;
+#if defined(_MSC_VER) && _MSC_VER < 1900
+            __declspec(thread) static char buffer[sizeof(GenericValue)];
+            return *new (buffer) GenericValue();
+#else
+#if defined(RAPIDJSON_HAS_CXX11)
+            thread_local static GenericValue buffer;
+#else
+            __thread static GenericValue buffer;
```

Comment:
`__thread` is not C++ standard I think this need to be only used in specific compilers

Reply:
Yes, it's GCC and Clang extension. I've changed the code to restrict the compiler to GCC and Clang for `__thread` and added fallback code to use regular static storage otherwise.

@miloyip